sympy/core/relational.py

```diff
@@ -1342,6 +1343,10 @@ def is_eq(lhs, rhs):
                 return retval
 
     retval = _eval_is_eq(lhs, rhs)
+
+    if isinstance(lhs, Set) and not isinstance(rhs, Set) or not isinstance(lhs, Set) and isinstance(rhs, Set):
+        retval = False
```

Comment:
A better fix would be to add `_eval_Is_eq` handlers in sets/handlers. We can have a handler for `Set, Basic` that always returns False and then a handler for `Set, Set` that returns `None`.

Reply:
Yes , thats what I tried at first but the two modules are so closely linked that their circular import issue couldn't be solved.

Reply:
There will be some way to solve the circular import. It just sometimes involves moving some imports around in other places

Reply:
I'll give it another try